presto-openapi/src/main/resources/cluster.yaml

```diff
@@ -0,0 +1,139 @@
+    get:
+      summary: Retrieve cluster memory pool information
+      description: Retrieves information about the memory pools in the Presto cluster.
+      responses:
+        '200':
+          description: Cluster memory pool information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ClusterMemoryPoolInfo'
+  /v1/cluster/workerMemory:
+    get:
+      summary: Retrieve worker memory information
+      description: Retrieves information about the memory usage of workers in the Presto cluster.
+      responses:
+        '200':
+          description: Worker memory information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WorkerMemoryInfo'
+  /v1/cluster/ttl:
+    get:
+      summary: Retrieve cluster TTL information
+      description: Retrieves time-to-live (TTL) information for the Presto cluster.
+      responses:
+        '200':
+          description: Cluster TTL information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ClusterTtl'
+components:
+  schemas:
+    ClusterStats:
+      type: object
+      properties:
+        runningQueries:
+          type: integer
+          description: The number of running queries
+        blockedQueries:
+          type: integer
+          description: The number of blocked queries
+        queuedQueries:
+          type: integer
+          description: The number of queued queries
+        activeWorkers:
+          type: integer
+          description: The number of active workers
+        runningDrivers:
+          type: integer
+          description: The number of running drivers
+        runningTasks:
+          type: integer
+          description: The number of running tasks
+        reservedMemory:
+          type: number
+          format: double
+          description: The amount of reserved memory
+        totalInputRows:
+          type: integer
+          description: The total number of input rows
+        totalInputBytes:
+          type: integer
+          description: The total number of input bytes
+        totalCpuTimeSecs:
+          type: integer
+          description: The total CPU time in seconds
+        adjustedQueueSize:
+          type: integer
+          description: The adjusted queue size
+    ClusterMemoryPoolInfo:
+      type: object
+      properties:
+        pools:
+          type: array
+          items:
+            type: object
+            properties:
+              name:
+                type: string
+              reserved:
+                type: number
+                format: double
+              free:
+                type: number
+                format: double
+              total:
+                type: number
+                format: double
+    WorkerMemoryInfo:
+      type: object
+      properties:
+        workers:
```

Comment:
So the output of the `getWorkerMemoryInfo` method which generates the response for this API is a `Map<String, MemoryInfo>`, so this schema need to be updated to reflect that - it isn't an array. You'll also need to add the `MemoryInfo`, `DataSize`, `MemoryPoolId`, and `MemoryPoolInfo` schema objects.